src/Cli/dotnet/ToolPackage/ToolPackageInstance.cs

```diff
@@ -97,12 +96,10 @@ private IReadOnlyList<RestoredCommand> GetCommands()
                 }
 
                 // Currently only "dotnet" commands are supported
-                commands.Add(new RestoredCommand(
+                return new RestoredCommand(
                     new ToolCommandName(configuration.CommandName),
                     "dotnet",
-                    LockFileRelativePathToFullFilePath(entryPointFromLockFile.Path, library)));
-
-                return commands;
+                    LockFileRelativePathToFullFilePath(entryPointFromLockFile.Path, library));
```

Comment:
I'm just asking this to see if the check around `Command is not null` makes sense. If this method always returns a `RestoredCommand`, that means the lazy instance always has a value. That means `_command.Value` in `Command` property getter always has a value. This means that accessing that property cannot be null.

So, I know the checks before where if the command list had any entries, but it doesn't seem like that was possible to not have any, unless the entries were removed or cleared before (not sure if that happened). Do you think that the check for nullability of `Command` is even necessary at this point?

Reply:
I'm not an expert with Lazy, and perhaps that check isn't necessary, but I thought it was possible for GetCommand() to throw an exception before it returns a RestoredCommand. If that exception is then caught, perhaps accessing Command would return null because it had never been assigned a value? I think `is not null` is semantically equivalent to the previous check for an empty list, so I'd _slightly_ prefer to keep it, but if you think it's unnecessary, I wouldn't mind removing it.